---
title: "collapsetaxa_eDNAindex"
author: "Eily Allan"
date: "11/20/2021"
output: html_document
---

After merging all the ASV tables and hash keys, let's pair them together with taxonomy - and also go from ASV tables to taxa tables and collapse by taxa. Then we can transform everything to eDNA index (at ASV and taxa level) to do some more plotting and exploring. 

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(tidyverse)
library(vegan)
library(reshape2)
```

## Read everything in and combine 

```{r read in files}

# Read the merged ASV tables
COI.ASV.table <- read_csv(paste0(here("Output", "dada2_output"), "/", "20211230.combined.COI.ASV.table.csv"))
MiFish.ASV.table <- read_csv(paste0(here("Output", "dada2_output"), "/", "20211230.combined.MiFish.ASV.table.csv"))
MiMammal.ASV.table <- read_csv(paste0(here("Output", "dada2_output"), "/", "20211230.combined.MiMammal.ASV.table.csv"))

# Read in the classification output for each run 
# the files written to the general folder (not run specific) should have everything classified from all runs because they are our "databases" to use for future runs - we want to use the "DATE.MARKER.all.gs.previous.hashes.annotated.rds" file 
COI.annotations <- readRDS(file=paste0(here("Output","classification_output"),"/COI.all.gs.previous.hashes.annotated.rds"))
MiFish.annotations <- readRDS(file=paste0(here("Output","classification_output"),"/MiFish.all.gs.previous.hashes.annotated.rds"))
MiMammal.annotations <- readRDS(file=paste0(here("Output","classification_output"),"/MiMammal.all.gs.previous.hashes.annotated.rds"))

all.metadata <- read.csv(here("Input", "sequencing_metadata_files", "master_sequencing_datasheet_20211026.csv"))

```

## Add taxonomy and make tables 

```{r add tax and collapse}
# this taxonomy file is the "good" annotations to species and genus level so let's get rid of the other junk for right now 
MiMammal.simple.annotations <- MiMammal.annotations %>% select(representative, taxon, genus, species) %>% rename(Hash = representative)
MiMammal.ASV.taxon <- MiMammal.ASV.table %>% 
  left_join(MiMammal.simple.annotations, by = "Hash") %>% 
  select(-Locus)

# ok so that has everything.. let's remove (and save) ASVs that don't have an annotation 
MiMammal.ASV.no.taxon <- MiMammal.ASV.taxon[is.na(MiMammal.ASV.taxon$taxon),]
# and same - keep things that do have an annotation (reminder to species or genus level so "good" annotation)
MiMammal.ASV.yes.taxon <- MiMammal.ASV.taxon[! is.na(MiMammal.ASV.taxon$taxon),]

# compare number of ASVs and reads that are assigned to species and genus (or not)
MiMammal.n.ASV.no.taxon <- length(unique(MiMammal.ASV.no.taxon$Hash))
MiMammal.n.reads.ASV.no.taxon <- sum(MiMammal.ASV.no.taxon$nReads)
MiMammal.n.ASV.yes.taxon <- length(unique(MiMammal.ASV.yes.taxon$Hash))
MiMammal.n.reads.ASV.yes.taxon <- sum(MiMammal.ASV.yes.taxon$nReads)
MiMammal.percent.ASV.no.taxon <- MiMammal.n.ASV.no.taxon/(MiMammal.n.ASV.no.taxon+MiMammal.n.ASV.yes.taxon)*100 
MiMammal.percent.reads.ASV.no.taxon <- MiMammal.n.reads.ASV.no.taxon/(MiMammal.n.reads.ASV.no.taxon+MiMammal.n.reads.ASV.yes.taxon)*100 
# ok so 90% of ASVs are not to genus or species BUT only 30% of *reads* are not assigned to genus or species
# that seems ok for now - we can go back later and use the whole classification and see if we can get more info (family, order, etc.)

# let's take only the ASVs that are annotated to something and collapse taxa (add reads together of ASVs that go to the same taxon)
MiMammal.by.taxa <- MiMammal.ASV.yes.taxon %>% 
  select(-Hash) %>% # we don't need the hash identifier anymore
  group_by(Sample_name, taxon) %>% # for each sample that has multiple asvs that assign to the same taxa...
  summarise(tot = sum(nReads))

MiMammal.by.taxa.pivot <- MiMammal.by.taxa %>% 
  pivot_wider(names_from = Sample_name, values_from =  tot) %>% # switch to rows (taxa) by columns (samples)
  column_to_rownames(var="taxon")

# we can also do this at the genus level 
MiMammal.by.genus <- MiMammal.ASV.yes.taxon %>% 
  select(-Hash) %>% # we don't need the hash identifier anymore
  group_by(Sample_name, genus) %>% # for each sample that has multiple asvs that assign to the same taxa...
  summarise(tot = sum(nReads))

MiMammal.by.genus.pivot <- MiMammal.by.genus %>% 
  pivot_wider(names_from = Sample_name, values_from =  tot) %>% # switch to rows (taxa) by columns (samples)
  column_to_rownames(var="genus")

# ok but we really should be using eDNA index instead of reads to be comparing things (LATER LET'S GO BACK AND DO BETTER)
source(here("functions", "eDNAindex.R"))
MiMammal.ASV.index <-  eDNAindex(MiMammal.ASV.table, Sample_name, Hash, nReads) # here use everything whether it was assigned taxonomy or not
MiMammal.taxa.index <- eDNAindex(MiMammal.by.taxa, Sample_name, taxon, tot)
MiMammal.genus.index <- eDNAindex(MiMammal.by.genus, Sample_name, genus, tot)

MiMammal.genus.index.pivot <- MiMammal.genus.index %>% 
  pivot_wider(names_from = Sample_name, values_from =  Normalized.reads) %>% # switch to rows (taxa) by columns (samples)
  column_to_rownames(var="genus")

# let's be really bad and just pool biological replicates right now....
MiMammal.by.genus.bio <- MiMammal.by.genus %>% 
  separate(Sample_name, into=c("marker","date","creek","site","bio_rep"), remove = FALSE) %>% 
  unite("original_sample", "marker","date","creek","site")

MiMammal.by.genus.pooled.bio <- MiMammal.by.genus.bio %>% 
  separate(Sample_name, into=c("marker","date","creek","site","bio_rep"), remove = FALSE) %>% 
  unite("original_sample", "marker","date","creek","site") %>% 
  group_by(original_sample, genus) %>% # for each sample that has multiple asvs that assign to the same taxa...
  summarise(tot = sum(tot)) 

MiMammal.genus.bio.index <- eDNAindex(MiMammal.by.genus.bio, original_sample, genus, tot)
MiMammal.genus.bio.index2 <- eDNAindex(MiMammal.by.genus.bio, original_sample, genus, tot, bio_rep)
MiMammal.genus.pooled.bio.index <- eDNAindex(MiMammal.by.genus.pooled.bio, original_sample, genus, tot)

MiMammal.genus.pooled.bio.index.pivot <- MiMammal.genus.pooled.bio.index%>% 
  pivot_wider(names_from = original_sample, values_from =  Normalized.reads) %>% # switch to rows (taxa) by columns (samples)
  column_to_rownames(var="genus")

```

## BC distances, plots, etc. 

```{r visualizing}

# damn so many things to chose from... 
taxon.for.anova <- MiMammal.genus.index %>% 
  rename(taxon = genus) 
  
source(here("functions", "tibble_to_matrix.R"))
all.distances.full <- tibble_to_matrix (taxon.for.anova)

as_tibble(subset(melt(as.matrix(all.distances.full)))) -> all.distances.melted
# Any mjor screw ups
summary(is.na(all.distances.melted$value))

# remove kanagroo 
all.distances.melted <- all.distances.melted[! str_detect(all.distances.melted$Var1,"Kangaroo"),]
all.distances.melted <- all.distances.melted[! str_detect(all.distances.melted$Var2,"Kangaroo"),]

## add on run and well
all.distances.melted <- all.distances.melted %>% 
  rename(Sample_name = Var1)

metadata.layout <- all.metadata %>% 
  select(Sequencing.run, Sample_ID, Well) %>% 
  rename(Sample_name = Sample_ID)

all.distances.melted <- all.distances.melted %>% 
  left_join(metadata.layout, by= "Sample_name") %>% 
  rename(Var1 = Sample_name) %>% 
  rename(var1run = Sequencing.run) %>% 
  rename(var1well = Well) 

all.distances.melted <- all.distances.melted %>% 
  rename(Sample_name = Var2)

all.distances.melted <- all.distances.melted %>% 
  left_join(metadata.layout, by= "Sample_name") %>% 
  rename(Var2 = Sample_name) %>% 
  rename(var2run = Sequencing.run) %>% 
  rename(var2well = Well) 


# separate big sample names
all.distances.melted <-
  all.distances.melted %>%
  separate (Var1, into = c("Marker1", "Date1", "Creek1", "Site1", "Bottle1"), remove = FALSE) %>%
  separate (Var2, into = c("Marker2", "Date2", "Creek2", "Site2", "Bottle2"), remove = FALSE) 

# fix the hellhole of "2TR2" etc.
all.distances.melted <-
  all.distances.melted %>% 
  mutate(Bottle1 = str_replace(Bottle1, "^1$", "1TR1")) %>% 
  mutate(Bottle2 = str_replace(Bottle2, "^1$", "1TR1")) %>% 
  mutate(Bottle1 = str_replace(Bottle1, "^2$", "2TR1")) %>%
  mutate(Bottle2 = str_replace(Bottle2, "^2$", "2TR1")) %>%
  mutate(Bottle1 = str_replace(Bottle1, "^3$", "3TR1")) %>%
  mutate(Bottle2 = str_replace(Bottle2, "^3$", "3TR1")) %>% 
  separate(Bottle1, into = c("Bottle1", "TechRep1"), sep= -3, remove=TRUE) %>% 
  separate(Bottle2, into = c("Bottle2", "TechRep2"), sep= -3, remove=TRUE)


all.distances.melted %>%
  mutate ( Date.Creek.Site1 = paste0(Date1, Creek1, Site1),
           Date.Creek.Site2 = paste0(Date2, Creek2, Site2),
           Date.Creek1 = paste0(Date1, Creek1),
           Date.Creek2 = paste0(Date2, Creek2),
           Distance.type = case_when( Date.Creek.Site1 == Date.Creek.Site2 & Bottle1 == Bottle2 ~ "Tech.replicates",
                                      Date.Creek.Site1 == Date.Creek.Site2 ~ "Biol.replicates",
                                      Date.Creek1 == Date.Creek2 ~ "Same.datecreek.Diff.site",
                                      Date1 == Date2 & Creek1 != Creek2 ~ "Same.date.Diff.creek",
                                      Creek1 == Creek2 & Date1 != Date2 ~ "Same.creek.different.date",
                                      Creek1 != Creek2 & Date1 != Date2 ~ "Different.creek.date",
                                      TRUE ~ "FIX"
                                     )) %>%
  dplyr::select(Sample1 = Var1, Sample2 = Var2 , value , Distance.type, var1run, var1well, var2run, var2well) %>%
  filter (Sample1 != Sample2) -> all.distances.to.plot

# remove when 1 = 2 and then 2 = 1 (remove duplicates)
#all.distances.to.plot$paired <- paste0(all.distances.to.plot$Sample1, all.distances.to.plot$Sample2)
#all.distances.to.plot <- all.distances.to.plot %>% distinct(paired, .keep_all = TRUE)

# Checking all went well
sapply(all.distances.to.plot, function(x) summary(is.na(x))) # good boi
all.distances.to.plot$Distance.type <- all.distances.to.plot$Distance.type  %>% fct_relevel("Tech.replicates", "Biol.replicates", "Same.datecreek.Diff.site", "Same.date.Diff.creek", "Same.creek.different.date", "Different.creek.date")
  
ggplot (all.distances.to.plot) +
  geom_histogram (aes (fill = Distance.type, x = value, after_stat(ndensity)), position = "dodge",  alpha = 0.9, bins = 50) +
  facet_wrap( ~ Distance.type) +
  labs (x = "Pairwise dissimilarity", y = "density" ,
        Distance.type = "Distance") +
    guides (fill = "none")

#ggsave(file="/Users/elizabethandruszkiewicz/GoogleDrive/UW/GitHub/NextGenNEPA_EA/Output/MiMammal.allruns.asvlevel.index.visual.anova.png", dpi = "retina")


diffcreekdate <- all.distances.to.plot %>% 
  filter(Distance.type == "Different.creek.date")

didwefuckup <- all.distances.to.plot %>% 
  filter(var1run == var2run) %>% 
  filter(var1run == 2) 

didwefuckup2 <- didwefuckup %>% 
  group_by(var1well) %>% 
  top_n(1,-value) 
  




```

```{r mifish}

MiFish.simple.annotations <- MiFish.annotations %>% select(representative, taxon, genus, species) %>% rename(Hash = representative)
MiFish.ASV.taxon <- MiFish.ASV.table %>% 
  left_join(MiFish.simple.annotations, by = "Hash") %>% 
  select(-Locus)

# ok so that has everything.. let's remove (and save) ASVs that don't have an annotation 
MiFish.ASV.no.taxon <- MiFish.ASV.taxon[is.na(MiFish.ASV.taxon$taxon),]
# and same - keep things that do have an annotation (reminder to species or genus level so "good" annotation)
MiFish.ASV.yes.taxon <- MiFish.ASV.taxon[! is.na(MiFish.ASV.taxon$taxon),]

# compare number of ASVs and reads that are assigned to species and genus (or not)
MiFish.n.ASV.no.taxon <- length(unique(MiFish.ASV.no.taxon$Hash))
MiFish.n.reads.ASV.no.taxon <- sum(MiFish.ASV.no.taxon$nReads)
MiFish.n.ASV.yes.taxon <- length(unique(MiFish.ASV.yes.taxon$Hash))
MiFish.n.reads.ASV.yes.taxon <- sum(MiFish.ASV.yes.taxon$nReads)
MiFish.percent.ASV.no.taxon <- MiFish.n.ASV.no.taxon/(MiFish.n.ASV.no.taxon+MiFish.n.ASV.yes.taxon)*100 
MiFish.percent.reads.ASV.no.taxon <- MiFish.n.reads.ASV.no.taxon/(MiFish.n.reads.ASV.no.taxon+MiFish.n.reads.ASV.yes.taxon)*100 

```

